x-pack/plugins/apm/server/routes/traces/get_top_traces.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { sortBy } from 'lodash';
+import moment from 'moment';
+import {
+  SERVICE_NAME,
+  TRANSACTION_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { joinByKey } from '../../../common/utils/join_by_key';
+import { withApmSpan } from '../../utils/with_apm_span';
+import { Setup } from '../../lib/helpers/setup_request';
+import { getAverages, getCounts, getSums } from './get_transaction_group_stats';
+import { AgentName } from '../../../typings/es_schemas/ui/fields/agent';
+
+type Key = Record<'service.name' | 'transaction.name', string>;
+export interface TransactionGroup {
+  key: Key;
+  serviceName: string;
+  transactionName: string;
+  transactionType: string;
+  averageResponseTime: number | null | undefined;
+  transactionsPerMinute: number;
+  impact: number;
+  agentName: AgentName;
+}
+
+function getItemsWithRelativeImpact(
+  setup: Setup,
+  items: Array<{
+    sum?: number | null;
+    key: Key;
+    avg?: number | null;
+    count?: number | null;
+    transactionType?: string;
+    agentName?: AgentName;
+  }>,
+  start: number,
+  end: number
+) {
+  const values = items
+    .map(({ sum }) => sum)
+    .filter((value) => value !== null) as number[];
+
+  const max = Math.max(...values);
+  const min = Math.min(...values);
+
+  const duration = moment.duration(end - start);
+  const minutes = duration.asMinutes();
+
+  const itemsWithRelativeImpact = items.map((item) => {
+    return {
+      key: item.key,
+      averageResponseTime: item.avg,
+      transactionsPerMinute: (item.count ?? 0) / minutes,
+      transactionType: item.transactionType || '',
+      impact:
+        item.sum !== null && item.sum !== undefined
+          ? ((item.sum - min) / (max - min)) * 100 || 0
+          : 0,
+      agentName: item.agentName as AgentName,
+    };
+  });
+
+  return itemsWithRelativeImpact;
+}
+
+export function getTopTraces({
+  environment,
+  kuery,
+  setup,
+  searchAggregatedTransactions,
+  start,
+  end,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  searchAggregatedTransactions: boolean;
+  start: number;
+  end: number;
+}) {
+  return withApmSpan('get_top_traces', async () => {
+    const params = {
+      environment,
+      kuery,
+      setup,
+      searchAggregatedTransactions,
+      start,
+      end,
+    };
+
+    const [counts, averages, sums] = await Promise.all([
+      getCounts(params),
+      getAverages(params),
+      getSums(params),
+    ]);
```

Comment:
@dgieselaar Does it (still) make sense to do this in three separate requests? All three requests share the same query, they just apply a different sub agg. In that case I'd expect a single request to ES to put less load on the system, since the same documents don't have to be found and returned three times. 

Do you recall why we did this in the first place?

Reply:
Parallelisation for top traces view, presumably. Running three aggs in separate requests requires more cumulative CPU on ES' side, but it is (a little bit) faster in terms of response time. We can remove it AFAIC, like we did for the service inventory.

Reply:
Okay, unless the gains are significant I think we should remove it to simplify the code. fyi @kpatticha

Reply:
@sqren @dgieselaar can you please point me to the service inventory so I could take a look 🙏

Reply:
Service inventory is `GET /internal/apm/services` (and related requests). But I don't think you need to look at that. 
What I'm suggesting is instead of doing three requests `getCounts`, `getAverages` and `getSums` we do a single request.

Reply:
@kpatticha I'm available for zoom if needed.